src/containers/Timesheet/Timesheet.tsx

```diff
@@ -1,14 +1,54 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { useDropzone } from "react-dropzone";
 import { Container, Box, Typography } from "@mui/material";
 import Accordion from "@mui/material/Accordion";
 import AccordionSummary from "@mui/material/AccordionSummary";
 import AccordionDetails from "@mui/material/AccordionDetails";
 import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
 import { UIConstants } from "@constants/UIConstants";
 import { Textarea } from "@components/Textarea/Textarea";
+import {
+    ThumbInner,
+    Thumb,
+    Image,
+    Dropzone,
+    ThumbsContainer,
+} from "./Timesheet.styles";
 import { CustomButton } from "../../components/CustomButton/CustomButton";
 
 export const Timesheet: React.FC = () => {
+    const [files, setFiles] = useState([]);
+    const { getRootProps, getInputProps } = useDropzone({
+        accept: {
+            "image/*": [],
+        },
+        onDrop: (acceptedFiles) => {
+            setFiles(
+                acceptedFiles.map((file) =>
+                    Object.assign(file, {
+                        preview: URL.createObjectURL(file),
+                    }),
+                ),
+            );
+        },
+    });
+    const thumbs = files.map((file) => (
+        <Thumb>
+            <ThumbInner>
+                <Image
+                    src={file.preview}
+                    alt="Preview"
+                    onLoad={() => {
+                        URL.revokeObjectURL(file.preview);
+                    }}
+                />
+            </ThumbInner>
+        </Thumb>
+    ));
```

Comment:
Do not include the component inside another component. Extract it into a separate component and pass the required fields as props and return the result.